spec/solargraph-rails/model_spec.rb

```diff
@@ -93,4 +93,28 @@ class Person < ActiveRecord::Base
       expect(pin.parameters.first.name).to eq('min_height')
     end
   end
+
+  it 'does not generate methods for variable named scope' do
+    load_string 'app/models/person.rb',
+                <<-RUBY
+                class Person < ActiveRecord::Base
+                  def some_method
+                    scope = Person.where(id: 0)
+                    scope.count
+                  end
+
+                  scope :taller_than,
+                        ->(min_height) { where('height > ?', min_height) }
+                end
+                RUBY
+
+    assert_class_method(
```

Comment:
in here we probably want to assert that no wrong scope was created, instead of doing assertions on `taller_than`

Reply:
I wasn't able to find a way to do that, I don't think a scope is actually created due to the plugin just crashing with this error
```
[WARN] undefined method `children' for nil:NilClass
/home/mathias/dev/solargraph-rails/lib/solargraph/rails/model.rb:31:in `block in process'
/home/mathias/dev/solargraph-rails/lib/solargraph/rails/walker.rb:69:in `try_match'
/home/mathias/dev/solargraph-rails/lib/solargraph/rails/walker.rb:48:in `block in traverse'
/home/mathias/dev/solargraph-rails/lib/solargraph/rails/walker.rb:48:in `each'
/home/mathias/dev/solargraph-rails/lib/solargraph/rails/walker.rb:48:in `traverse'
```

Seems like the `scope = Person.where(id: 0)` is unrelated and the crash will happen even if you just do `scope.count`

Reply:
I expected to be able to use a expect {..}.not_to raise_error but seems like all errors are caught and then logged. Maybe there is a way to check if an error was logged?

Reply:
I see. That's because solaragraph-rails catches errors in its internal plugins so that other plugins won't crash. In your case the test will actually fail with your current assertion, because the scope call below won't be parsed